docs/user_guide/preemption.md

```diff
@@ -0,0 +1,252 @@
+
+
+### PriorityClass
+
+In this example, we will demonstrate the configuration of `yunikorn.apache.org/allow-preemption` using PriorityClass and observe its effect. The default value for this configuration is set to `true`.
+
+```bash
+apiVersion: scheduling.k8s.io/v1
+kind: PriorityClass
+metadata:
+  name: preemption-not-allow
+  annotations:
+    "yunikorn.apache.org/allow-preemption": "false"
+value: 0
+```
+
+We will deploy 8 pods with a resource requirement of 1 to `queue-1`, `queue-2`, and `queue-3`, respectively. We will deploy to `queue-1` and `queue-2` first, followed by a few seconds delay before deploying to `queue-3`. This ensures that the resource usage in `queue-1` and `queue-2` will be greater than that in `queue-3`, depleting all resources in the parent queue and triggering preemption.
+
+| Queue                    | Max Resource | Guaranteed Resource | Policy                      |
+| ------------------------ | ------------ | ------------------- | --------------------------- |
+| `priority-class`         | 16           | -                   |                             |
+| `priority-class.queue-1` | 8            | 3                   | `"allow-preemption": true`  |
+| `priority-class.queue-2` | 8            | 3                   | `"allow-preemption": false` |
+| `priority-class.queue-3` | 8            | 3                   | `"allow-preemption": true`  |
+
+Result:
+
+When preemption is triggered, `queue-3` will start searching for a victim. However, since `queue-2` is set with allow-preemption as false, the resources of `queue-1` will be preempted.
+
+Please note that setting `yunikorn.apache.org/allow-preemption` is a strong recommendation but does not guarantee the lack of preemption. When this flag is set to `false`, it moves the Pod to the back of the preemption list, giving it a lower priority for preemption compared to other Pods. However, in certain scenarios, such as when no other preemption options are available, Pods with this flag may still be preempted.
+
+For example, even with `allow-preemption` set to `false`, DaemonSet pods can still be preempted. Additionally, if an application in `queue-1` has a higher priority than one in `queue-3`, the application in `queue-2` will be preempted because an application can never preempt another application with a higher priority. In such cases where no other preemption options exist, the `allow-preemption` flag may not prevent preemption.
+
+
+| Queue                  | Resource before preemption | Resource after preemption              |
+| ---------------------- | -------------------------- | -------------------------------------- |
+| `priority-class.queue-1` | 8 (victim)                 | 5                                      |
+| `priority-class.queue-2` | 8                          | 8                                      |
+| `priority-class.queue-3` | 0                          | 3 (guaranteed resources are satisfied) |
```

Comment:
Formatting issue again - use `(guaranteed minimum)`. Also, the queue names probably need to be shortened in order to fit without wrapping.